libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -0,0 +1,614 @@
+                process_packet();
+            } else {
+                debug("crc expected:%x got:%x", (unsigned)expected_crc, (unsigned)_parsed_msg.crc16);
+            }
+            reset_parser = true;
+            break;
+        }
+
+        // handle reset of parser
+        if (reset_parser) {
+            _parsed_msg.state = ParseState::WAITING_FOR_HEADER_LOW;
+            _msg_buff_len = 0;
+        }
+    }
+}
+
+// process successfully decoded packets held in the _parsed_msg structure
+void AP_Mount_Siyi::process_packet()
+{
+    // flag to warn of unexpected data buffer length
+    bool unexpected_len = false;
+
+    // process packet depending upon command id
+    switch ((SiyiCommandId)_parsed_msg.command_id) {
+
+    case SiyiCommandId::ACQUIRE_FIRMWARE_VERSION: {
+        if (_parsed_msg.data_bytes_received != 12) {
+            unexpected_len = true;
+            break;
+        }
+        _got_firmware_version = true;
+
+        // display camera firmware version
+        debug("Mount: SiyiCam fw:%u.%u.%u",
+              (unsigned)_msg_buff[_msg_buff_data_start+1],      // firmware major version
+              (unsigned)_msg_buff[_msg_buff_data_start+2],      // firmware minor version
+              (unsigned)_msg_buff[_msg_buff_data_start+3]);     // firmware revision
+
+        // display gimbal info to user
+        gcs().send_text(MAV_SEVERITY_INFO, "Mount: Siyi fw:%u.%u.%u",
+                (unsigned)_msg_buff[_msg_buff_data_start+5],    // firmware major version
+                (unsigned)_msg_buff[_msg_buff_data_start+6],    // firmware minor version
+                (unsigned)_msg_buff[_msg_buff_data_start+7]);   // firmware revision
+
+        // display zoom firmware version
+        debug("Mount: SiyiZoom fw:%u.%u.%u",
+              (unsigned)_msg_buff[_msg_buff_data_start+9],      // firmware major version
+              (unsigned)_msg_buff[_msg_buff_data_start+10],     // firmware minor version
+              (unsigned)_msg_buff[_msg_buff_data_start+11]);    // firmware revision
+        break;
+    }
+
+    case SiyiCommandId::HARDWARE_ID:
+        // unsupported
+        break;
+
+    case SiyiCommandId::AUTO_FOCUS:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        debug("AutoFocus:%u", (unsigned)_msg_buff[_msg_buff_data_start]);
+        break;
+
+    case SiyiCommandId::MANUAL_ZOOM_AND_AUTO_FOCUS: {
+        if (_parsed_msg.data_bytes_received != 2) {
+            unexpected_len = true;
+            break;
+        }
+        const double zoom_mult = UINT16_VALUE(_msg_buff[_msg_buff_data_start+1], _msg_buff[_msg_buff_data_start]) * 0.1;
+        debug("ZoomMult:%4.1f", (double)zoom_mult);
+        break;
+    }
+
+    case SiyiCommandId::MANUAL_FOCUS:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
```

Comment:
This is probably good diagnostics during dev, but do we want to spend the bytes when we actually have this in-tree to incrememnt this counter?  We might save some bytes if we consolidate all of these 1-bytes-received cases into one.

Reply:
Yes, I've added, "#if AP_MOUNT_SIYI_DEBUG" around all these to save a little flash.